lang/intrinsic.md

```diff
@@ -140,7 +152,93 @@ impl<M: Memory> Machine<M> {
 
         self.mem.deallocate(ptr, size, align)?;
 
-        ret(())
+        ret(ExecutionStatus::Finished)
     }
 }
 ```
+
+And finally the intrinsics for spawning and joining threads.
+
+```rust
+impl<M: Memory> Machine<M> {
+    fn eval_intrinsic(
+        &mut self,
+        Intrinsic::Spawn: Intrinsic,
+        arguments: List<Value<M>>,
+        ret_place: Option<Place<M>>,
+    ) -> NdResult<ExecutionStatus> {
+        if arguments.len() != 1 {
+            throw_ub!("invalid number of arguments for `Intrinsic::Spawn`");
+        }
+
+        let Value::Ptr(ptr) = arguments[0] else {
+            throw_ub!("invalid first argument to `Intrinsic::Spawn`");
+        };
+
+        let mut funcs = self.fn_addrs.iter().filter(|(_, fn_addr)| *fn_addr == ptr.addr);
+        let Some((func_name, _)) = funcs.next() else {
+            throw_ub!("invalid first argument to `Intrinsic::Spawn`, calling an address where there is no function");
+        };
+        let func = self.prog.functions[func_name];
```

Comment:
This looks like we want a helper function to obtain a `Function` from a pointer, to share this code with `Call`.

Reply:
Where should I define such helper functions?

Reply:
`machine.md` seems like a good place for now